tools/clang/test/HLSLFileCheck/hlsl/workgraph/multi_dim_array_record.hlsl

```diff
@@ -14,9 +14,9 @@
 // CHECK-DAG: [[ORH:%[0-9]+]] = call %dx.types.NodeRecordHandle @dx.op.allocateNodeOutputRecords(i32 238, %dx.types.NodeHandle [[ANOH]], i32 1, i1 false)
 // CHECK-DAG: [[AORH:%[0-9]+]] = call %dx.types.NodeRecordHandle @dx.op.annotateNodeRecordHandle(i32 251, %dx.types.NodeRecordHandle [[ORH]], %dx.types.NodeRecordInfo { i32 70, i32 32768 })
 // CHECK-DAG: [[IRP:%[0-9]+]] = call %[[RECORD:struct\.MyRecord[^ ]*]] addrspace(6)* @dx.op.getNodeRecordPtr.[[RECORD]](i32 239, %dx.types.NodeRecordHandle [[AIRH]], i32 0)
-// CHECK-DAG: getelementptr {{(inbounds )?}}%[[RECORD]], %[[RECORD]] addrspace(6)* [[IRP]], i32 0, i32 0, i32 [[TID]], i32 0
+// CHECK-DAG: getelementptr {{(inbounds )?}}%[[RECORD]], %[[RECORD]] addrspace(6)* [[IRP]], i32 0, i32 0, i32 [[TID]]
```

Comment:
Looks like this might have become a multi-level GEP?  I think this test change might violate the purpose of this test.
```
// Ensure multi-dim array produced by lowering record struct with array of
// vector has GEP instructions merged to prevent invalid GEP with array type.
```

We might have to do `MergeGepUse` after the call to `ReplaceUsesForLoweredUDTImpl` in `ReplaceUsesForLoweredUDT`.

Reply:
Updated.